### ISSUE TITLE
TypeError: __init__() missing 1 required positional argument: 'req_tracker'

### ISSUE THREAD

**Issue by MythicManiac** [suutari/prequ#28]
Running prequ update yields the following result:

```
root@5b55f2c59a74:/app# prequ update
*** Compiling requirements.txt
Traceback (most recent call last):
  File "/usr/local/bin/prequ", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/prequ/scripts/update.py", line 17, in main
    ctx.invoke(compile.main, verbose=verbose, silent=silent)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/prequ/scripts/compile.py", line 26, in main
    compile(ctx, verbose, silent, check)
  File "/usr/local/lib/python3.7/site-packages/prequ/scripts/compile.py", line 48, in compile
    do_one_file(ctx, conf, label, compile_opts)
  File "/usr/local/lib/python3.7/site-packages/prequ/scripts/compile.py", line 63, in do_one_file
    **compile_opts)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/prequ/scripts/compile_in.py", line 174, in cli
    results = resolver.resolve(max_rounds=max_rounds)
  File "/usr/local/lib/python3.7/site-packages/prequ/resolver.py", line 120, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "/usr/local/lib/python3.7/site-packages/prequ/resolver.py", line 218, in _resolve_one_round
    for dep in self._iter_dependencies(best_match):
  File "/usr/local/lib/python3.7/site-packages/prequ/resolver.py", line 307, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "/usr/local/lib/python3.7/site-packages/prequ/repositories/base.py", line 39, in get_dependencies
    return self._get_dependencies(ireq)
  File "/usr/local/lib/python3.7/site-packages/prequ/repositories/pypi.py", line 126, in _get_dependencies
    ireq, wheel_cache)
  File "/usr/local/lib/python3.7/site-packages/prequ/repositories/pypi.py", line 179, in _get_dependencies_with_wheel_cache
    build_isolation=False
TypeError: __init__() missing 1 required positional argument: 'req_tracker'
```

pip freeze
---

```
root@5b55f2c59a74:/app# pip freeze
aioredis==1.1.0
asgiref==2.3.2
asn1crypto==0.24.0
async-timeout==3.0.0
attrs==18.1.0
autobahn==18.6.1
Automat==0.6.0
certifi==2018.4.16
cffi==1.11.5
channels==2.1.1
channels-redis==2.2.1
chardet==3.0.4
click==6.7
constantly==15.1.0
cryptography==2.2.2
daphne==2.1.2
Django==2.0.3
django-environ==0.4.4
djangorestframework==3.8.2
h2==3.0.1
hiredis==0.2.0
hpack==3.0.0
hyperframe==5.1.0
hyperlink==18.0.0
idna==2.6
incremental==17.5.0
msgpack==0.5.6
Pillow==5.0.0
prequ==1.4.1
priority==1.3.0
psycopg2==2.7.4
psycopg2-binary==2.7.4
pyasn1==0.4.3
pyasn1-modules==0.2.1
pycparser==2.18
pyOpenSSL==18.0.0
pytz==2018.4
requests==2.18.4
service-identity==17.0.0
six==1.11.0
stripe==1.82.0
Twisted==18.4.0
txaio==2.10.0
urllib3==1.22
zope.interface==4.5.0
```

setup.cfg
---

```
[prequ]
annotate = yes
header = yes

requirements =
  django==2.0.7
  django-environ==0.4.5
  djangorestframework==3.8.2

requirements-dev =
  prequ==1.4.1
  pytest
  pytest-cov
  pytest-django
  model_mommy
  faker
```

ls
---

```
root@5b55f2c59a74:/app# ls
apps/      build.py       requirements-builder.txt      static/            static_src/
core/      manage.py      setup.cfg                     static_built/      var/
```

other details
---

- no requirement files that are configured in `setup.cfg` are present in the folder

**Comment by MythicManiac** [comment]
Downgrading to pip < 10 seems to fix the issue

**Comment by MythicManiac** [comment]
Also seems to work with pip 10.0.1

**Comment by suutari** [comment]
@MythicManiac Thanks for the bug report.

Gotta check the internal API changes they made in Pip 18 to make Prequ work with it. Pip no longer has a stable programmable API to use, so the only options are either (a) trying to keep up with their API changes on each new Pip release or (b) implementing the same functionality in Prequ (from scratch or by vendoring Pip). At the moment I'm still following the (a) route.

**Comment by suutari** [comment]
This should now be fixed (with PR #29) and the fix is released in Prequ 1.4.2. Can you confirm that Prequ 1.4.2 fixes the issue?

**Comment by MythicManiac** [comment]
@suutari can confirm, works with 1.4.2 👍 Thanks!
